packages/cpp/ArmoniK.Api.Client/source/submitter/ResultsClient.cpp

```diff
@@ -81,6 +81,77 @@ void ResultsClient::upload_result_data(const std::string &session_id, const std:
                                                                 status.error_message());
   }
 }
+
+void ResultsClient::delete_results(const std::string &session_id, const std::vector<std::string> &result_ids) {
+  if (result_ids.empty()) {
+    return;
+  }
+
+  ::grpc::ClientContext context;
+  armonik::api::grpc::v1::results::DeleteResultsDataRequest request;
+  armonik::api::grpc::v1::results::DeleteResultsDataResponse response;
+  *request.mutable_session_id() = session_id;
+  request.mutable_result_id()->Add(result_ids.begin(), result_ids.end());
+
+  auto status = stub->DeleteResultsData(&context, request, &response);
+  if (!status.ok()) {
+    throw armonik::api::common::exceptions::ArmoniKApiException("Unable to delete results " + status.error_message());
+  }
+}
+
+std::vector<armonik::api::grpc::v1::results::ResultRaw>
```

Comment:
I'd do an iterator to avoid having everything stored in a vector that would exceed RAM

Reply:
Too complex and pagination should solve the issue